### PR TITLE
Feature/journal detail cubit

### PR DESCRIPTION
**Pull request by AbdullahAhmedSoomro**
Journal Detail Cubit.

### CODE REVIEW

File: lib/cubits/journal_detail_cubit.dart [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import 'dart:async';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_form_builder/flutter_form_builder.dart';
+import 'package:star_book/cubits/cubit_state/cubit_state.dart';
+import 'package:star_book/cubits/journal_state.dart';
+import 'package:star_book/domain/repository/journal_repo.dart';
+
+class JournalDetailCubit extends Cubit<CubitState> {
+  final GlobalKey<FormBuilderState>? formKey;
```
Comment (hashirshoaeb) [inline lib/cubits/journal_detail_cubit.dart:10]: form key not needed

File: lib/cubits/journal_detail_cubit.dart [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  Future<void> getJournalByRange(
+      {required DateTime start, required DateTime end}) async {
+    emit(const LoadingState());
+    await journalRepo.getJournalByRange(start, end);
+  }
+
+  Future<void> deleteJournal({required String journalId}) async {
+    emit(const LoadingState());
+    await journalRepo.deleteJournal(journalId);
+  }
```
Comment (hashirshoaeb) [inline lib/cubits/journal_detail_cubit.dart:65]: not needed in details screen

File: lib/cubits/journal_detail_cubit.dart [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  /// Journal Stream by Day
+  Future<void> journalByDay$({required DateTime dateTime}) async {
+    journalStream = journalRepo.journalsByDay$(dateTime).listen((journals) {
+      journals.sort((a, b) {
+        return b.createdAt.compareTo(a.createdAt);
+      });
+      emit(LoadedState(
+          JournalState(journals: List.of(journals), selectedDate: dateTime)));
+    });
+  }
```
Comment (hashirshoaeb) [inline lib/cubits/journal_detail_cubit.dart:83]: not needed in details screen

File: lib/cubits/journal_state.dart [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:star_book/domain/models/journal/journal.dart';
+
+part 'journal_state.freezed.dart';
+
+@freezed
+class JournalState with _$JournalState {
+  factory JournalState({
+    required List<Journal> journals,
+    required DateTime selectedDate,
```
Comment (hashirshoaeb) [inline lib/cubits/journal_state.dart:10]:
```dart
required String journalId
```

File: lib/cubits/journal_detail_cubit.dart (hunk 2 above)
Comment (AbdullahAhmedSoomro) [reply lib/cubits/journal_detail_cubit.dart:65]: we need deleteJournal because we can delete journal from JournalDetailScreen

Comment (hashirshoaeb) [reply lib/cubits/journal_detail_cubit.dart:65]: oh yess

**Pull request merged**